lib/semian/lru_hash.rb

```diff
@@ -86,6 +94,8 @@ def clear_unused_resources
           resource.destroy
         end
       end
+    ensure
+      @lock.unlock if @lock.locked?
```

Comment:
Do we need this if statement if the ensure only happens after `try_lock`?

Reply:
Oops you're right, we won't get into the `begin` block unless the lock is locked.